go-controller/pkg/ovn/egressgw.go

```diff
@@ -222,7 +222,7 @@ func (oc *Controller) addGWRoutesForNamespace(namespace string, egress gatewayIn
 			if err != nil {
 				klog.Warningf("Unable to get port %s in cache for SNAT rule removal", logicalPort)
 			} else {
-				if err = deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, portInfo.ips); err != nil {
+				if err = deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, oc.watchFactory, portInfo.ips); err != nil {
```

Comment:
@tssurya so now I'm thinking, if we're passing two things into these functions that are members of the `oc` struct, should we just make the functions struct functions and reduce the number of args?

Reply:
yeah I wish, the problem is code that uses this from egressIPs is using the `eIP` struct and I'd need to duplicate the function there if we make this a part of `oc` struct.

btw I am currently adding the logic to update the perPodSNAT's for a l3gateway change per our last discussion.

Reply:
@dcbw if we prefer readability over code duplication I can redo the function in egressIPs part.

Reply:
nah, it's fine how it is now if it means code duplication.